app/lib/efile/ny/it201.rb

```diff
@@ -3,63 +3,68 @@ module Ny
     class It201
       attr_reader :lines
 
-      def initialize(year:, filing_status:, claimed_as_dependent:, dependent_count:, input_lines:, it227:)
+      def initialize(year:, filing_status:, claimed_as_dependent:, dependent_count:, input_lines:, it213:, it214:, it215:, it227:, value_access_tracker:)
         @year = year
 
         @filing_status = filing_status # single, married_filing_jointly, that's all we support for now
         @claimed_as_dependent = claimed_as_dependent # true/false
         @dependent_count = dependent_count # number
+        @value_access_tracker = value_access_tracker
+
         @lines = ActiveSupport::HashWithIndifferentAccess.new(input_lines)
+        @it213 = it213
+        @it214 = it214
+        @it215 = it215
         @it227 = it227
       end
 
       def calculate
-        set_line(:AMT_60E, @it227.calculate[:part2_line1])
-        set_line(:AMT_17, calculate_line_17)
-        set_line(:AMT_19, calculate_line_19)
-        set_line(:AMT_24, calculate_line_24)
-        set_line(:AMT_25, @lines[:AMT_4]&.value)
-        set_line(:AMT_32, calculate_line_32)
-        set_line(:AMT_33, calculate_line_33)
-        set_line(:AMT_34, calculate_line_34)
-        set_line(:AMT_35, calculate_line_35)
-        set_line(:AMT_36, @dependent_count)
-        set_line(:AMT_37, calculate_line_37)
-        set_line(:AMT_38, @lines[:AMT_37]&.value)
-        set_line(:AMT_39, calculate_line_39)
-        set_line(:AMT_40, calculate_line_40)
-        set_line(:AMT_43, calculate_line_43)
-        set_line(:AMT_44, calculate_line_44)
-        set_line(:AMT_46, calculate_line_46)
-        set_line(:AMT_47, calculate_line_47)
-        set_line(:AMT_47A, calculate_line_47a)
-        set_line(:AMT_48, calculate_line_48)
-        set_line(:AMT_49, calculate_line_49)
-        set_line(:AMT_52, calculate_line_52)
-        set_line(:AMT_54, calculate_line_54)
-        set_line(:AMT_54B, calculate_line_54b)
-        set_line(:AMT_58, calculate_line_58)
-        set_line(:AMT_61, calculate_line_61)
-        set_line(:AMT_63, calculate_line_63)
-        set_line(:AMT_65, calculate_line_65)
-        set_line(:AMT_67, calculate_line_67)
-        set_line(:AMT_69, calculate_line_69)
-        set_line(:AMT_69A, calculate_line_69a)
-        set_line(:AMT_70, calculate_line_70)
-        set_line(:AMT_73, calculate_line_72)
-        set_line(:AMT_73, calculate_line_73)
-        set_line(:AMT_76, calculate_line_76)
-        set_line(:AMT_77, calculate_line_77)
-        set_line(:AMT_78, calculate_line_78)
-        set_line(:AMT_78B, calculate_line_78b)
-        set_line(:AMT_80, calculate_line_80)
```

Comment:
If `set_line` took a block, we wouldn't have to mess with lambdas. Is that a better, more Ruby-ish API?

Reply:
blocks would be more natural than procs, but I dunno if we can do a `.source` of a block

though I'm kind of thinking we should just make a `calculate_line_x` for every line even if it's trivial, for consistency's sake --- `method(:calculate_line_4).source` produces something better to look at than sourcing a proc as it turns out

Reply:
Internet people say .source works for a block, so I might try that route.

Reply:
> I'm kind of thinking we should just make a calculate_line_x for every line even if it's trivial, for consistency's sake

I like this approach, and I like it more than the `block.source` hypothetical approach, too.